apis/spec/v1alpha2/servicebinding_types.go

```diff
@@ -0,0 +1,162 @@
+	Name string `json:"name,omitempty"`
+	// Selector is a query that selects the application or applications to bind the service to
+	Selector metav1.LabelSelector `json:"selector,omitempty"`
+	// Containers describes which containers in a Pod should be bound to
+	Containers []string `json:"containers,omitempty"`
+}
+
+// ServiceBindingServiceReference defines a subset of corev1.ObjectReference
+type ServiceBindingServiceReference struct {
+	// API version of the referent.
+	APIVersion string `json:"apiVersion"`
+	// Kind of the referent.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+	Kind string `json:"kind"`
+	// Name of the referent.
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+	Name string `json:"name"`
+}
+
+// ServiceBindingSecretReference defines a mirror of corev1.LocalObjectReference
+type ServiceBindingSecretReference struct {
+	// Name of the referent secret.
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+	Name string `json:"name"`
+}
+
+// EnvMapping defines a mapping from the value of a Secret entry to an environment variable
+type EnvMapping struct {
+	// Name is the name of the environment variable
+	Name string `json:"name"`
+	// Key is the key in the Secret that will be exposed
+	Key string `json:"key"`
+}
+
+// ServiceBindingSpec defines the desired state of ServiceBinding
+type ServiceBindingSpec struct {
+	// Name is the name of the service as projected into the application container.  Defaults to .metadata.name.
+	// +kubebuilder:validation:Pattern=`^[a-z0-9\-\.]*$`
+	// +kubebuilder:validation:MaxLength=253
+	Name string `json:"name,omitempty"`
+	// Type is the type of the service as projected into the application container
+	Type string `json:"type,omitempty"`
+	// Provider is the provider of the service as projected into the application container
+	Provider string `json:"provider,omitempty"`
+	// Application is a reference to an object
+	Application ServiceBindingApplicationReference `json:"application"`
+	// Service is a reference to an object that fulfills the ProvisionedService duck type
+	Service ServiceBindingServiceReference `json:"service"`
+	// Env is the collection of mappings from Secret entries to environment variables
+	Env []EnvMapping `json:"env,omitempty"`
+}
+
+// ServiceBindingStatus defines the observed state of ServiceBinding
+type ServiceBindingStatus struct {
+	// ObservedGeneration is the 'Generation' of the ServiceBinding that
+	// was last processed by the controller.
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// Conditions are the conditions of this ServiceBinding
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// Binding exposes the projected secret for this ServiceBinding
+	Binding *ServiceBindingSecretReference `json:"binding,omitempty"`
```

Comment:
I think we can directly use `*corev1.LocalObjectReference` here.

Reply:
I literately copied these types from https://github.com/k8s-service-bindings/spec/blob/master/internal/service.binding/v1alpha2/service_binding.go and it would be good if consume the structures in the form as the spec publishes them.